h5netcdf/core.py

```diff
@@ -125,28 +132,59 @@ def shuffle(self):
 
 NOT_A_VARIABLE = b'This is a netCDF dimension but not a netCDF variable.'
 
+class _object_names(Mapping):
+    def __init__(self, parent, object_cls):
+        self._parent = parent
+        self._object_cls = object_cls
+        self._objects = set()
```

Comment:
Right, I see. I think that was the reason I switched to `list` but I forgot. Let's use `OrderedDict`, it mimics better HDF5/netCDF.